.buildkite/pipeline.yml

```diff
@@ -12,4 +12,45 @@ steps:
     command: "make test"
     key: "tests"
     artifact_paths:
-      - "coverage.out"
\ No newline at end of file
+      - "coverage.out"
+
+  - label: ":sonarqube: Static Code Analysis"
+    env:
+      VAULT_SONAR_TOKEN_PATH: "kv/ci-shared/serverless/shared-analysis-token"
+    agents:
+      image: "docker.elastic.co/cloud-ci/sonarqube/buildkite-scanner:latest"
+    command:
+      - "buildkite-agent artifact download coverage.out ."
+      - "/scan-source-code.sh"
+    soft_fail: true
+    depends_on: "tests"
+
+  # Need go 1.20 in the serverless-docker-builder to enable this 
+  #- label: ":go: Build"
+  #  command: "make all"
```

Comment:
The agent used for this step is not the `serverless-docker-builder` but the one that is defined at the top of pipeline definition.

https://github.com/elastic/elasticsearch-k8s-metrics-adapter/blob/4541f679d297f79e2101e4ea5eec1fe066a24c87/.buildkite/pipeline.yml#L2

Reply:
ooh right 🤦‍♂️

Reply:
You can use `go-builder-buildkite-agent:1.20`, see https://github.com/elastic/ci-agent-images/pull/129/files.